compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -67,6 +67,47 @@ getStaticOrReifiedInputDims(OpBuilder &builder, Location loc, Value input,
   return success();
 }
 
+/// Returns the rank of the Value's type, and 0 if it is not a ShapedType.
+static int64_t getRank(Value v) {
+  auto type = dyn_cast<ShapedType>(v.getType());
+  if (type) {
+    return type.getRank();
+  }
+  return 0;
+}
+
+/// Method similar to `LinalgOp`s that concatenates shapes of all operands.
+static SmallVector<OpFoldResult>
+createFlatListOfOperandDims(OpBuilder &b, Location loc, Operation *op) {
+  SmallVector<OpFoldResult> res;
+  for (OpOperand &opOperand : op->getOpOperands()) {
+    for (auto dim : llvm::seq(getRank(opOperand.get()))) {
+      res.push_back(linalg::createFoldedDimOp(b, loc, opOperand.get(), dim));
+    }
+  }
+  return res;
+}
+
+/// Permutes the offset and size arrays by the result indexes of the provided
+/// affine map
+static SmallVector<Range> getPermutedRange(AffineMap permutation,
+                                           ArrayRef<OpFoldResult> offsets,
+                                           ArrayRef<OpFoldResult> sizes) {
+  auto one = IntegerAttr::get(IndexType::get(permutation.getContext()), 1);
+  assert(permutation.isProjectedPermutation() &&
+         "Affine map should be a projected permutation");
+  SmallVector<Range> output;
+  for (AffineExpr dimExpr : permutation.getResults()) {
+    int dim = cast<AffineDimExpr>(dimExpr).getPosition();
+    Range dimRange;
+    dimRange.offset = offsets[dim];
+    dimRange.size = sizes[dim];
+    dimRange.stride = one;
+    output.push_back(dimRange);
```

Comment:
optional: I'd just simplify it to `.push_back({offsets[dim], sizes[dim], one})`.